sql-plugin/src/main/340+/scala/com/nvidia/spark/rapids/shims/Spark340PlusShims.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import com.nvidia.spark.rapids.GpuCast
+
+import org.apache.spark.sql.catalyst.expressions.{Cast, Expression}
+
+trait Spark340PlusShims extends Spark331PlusShims {
+  override def ignoreTimeZone(e: Expression): Expression = e match {
+    case c: Cast if c.timeZoneId.nonEmpty && !c.needsTimeZone =>
```

Comment:
CastBase was a base class for Cast and AnsiCast. So we would need to add a match for AnsiCast in this solution

Reply:
I don't believe so, for 340+ `AnsiCast` was removed and merged into the Cast functionality (there's now a boolean arg baked into `Cast`)

Reply:
you are right, resolving.  apache/spark@5f0bae7af3c39919efd4bf68b1b6f467f0e1b20b